src/event/model/checkout_revision.rs

```diff
@@ -1,33 +1,15 @@
-use crate::event::model::CommonEventData;
+use super::event::AbstractEvent;
 use serde::Serialize;
 
 #[derive(Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct CheckoutRevision {
+    #[serde(flatten)]
+    pub abstract_event: AbstractEvent,
+
```

Comment:
This allows you to make the common fields explicit with a new struct, so:
- No copy-paste of common fields
- Less need for a dedicated constructor for each type using `CommonEventData` since you can pass `AbstractEvent` and use it in one go.